src/widgets.jl

```diff
@@ -362,7 +364,7 @@ function Options(view::Symbol,
 end
 
 function getoptions(options)
-    opts = OrderedDict()
+    opts = OrderedDict{String, eltype(options)}()
```

Comment:
Are these keys always Strings? Curious: why was this needed?

Reply:
Yes, the keys are always strings, all the elements are added by either
```julia
addoption!(opts, v::Union{Pair, Tuple{Any, Any}}) = opts[string(v[1])] = v[2]
```
or
```julia
addoption!(opts, v) = opts[string(v)] = v
```

Changing this line allows for increased type-specificity of signal( selection( input_array; multi=true))). Before, this signal always returned an Array{Any, 1}. Now it will instead return an array with the same type as the input_array. 

This enables a better compile-time optimisation of any function which takes this as an input.

Reply:
`opts = OrderedDict{Any, eltype(options)}() ` would also have worked. The type of the keys are not important, but the type of the value-container is.

Reply:
Also, the selection_slider has functionality for syncing to the nearest value if it does not correspond exactly to one of the possible values. To access this functionality, one must pass a 
```julia
syncselnearest = view == :SelectionSlider && typ <: Real && syncnearest

if syncnearest
```
statement. `typ` is defined by `typ=valtype(options.dict)`, which previously always were `Any`. Thus the selection_slider's syncnearest functionality was not possible to reach (`Any` is not a subtype of `Real`).

Reply:
Also, the selection_slider has functionality for syncing to the nearest value if it does not correspond exactly to one of the possible values. To access this functionality, one must pass a 
```julia
syncselnearest = view == :SelectionSlider && typ <: Real && syncnearest

if syncnearest
```
statement. `typ` is defined by `typ=valtype(options.dict)`, which previously always were `Any`. Thus the selection_slider's syncnearest functionality was not possible to reach (`Any` is not a subtype of `Real`).